### PR TITLE
use better defaults for march and cpu target

### PR DESCRIPTION
**Pull request by KristofferC**
Tried to adapt the stuff in  https://github.com/JuliaCI/julia-buildbot/blob/489ad6dee5f1e8f2ad341397dc15bb4fce436b26/master/inventory.py.
I'm a bit unsure what the `Sys.ARCH` actually are on the different platforms.

Does this look ok @staticfloat?

cc @terasakisatoshi

### CODE REVIEW

**Comment by KristofferC** [comment]
What happened with the arm builder... https://travis-ci.com/JuliaComputing/PackageCompilerX.jl/jobs/285129979

File: src/PackageCompilerX.jl [hunk 1]
```diff
@@ -19,18 +34,28 @@ all_stdlibs() = readdir(Sys.STDLIB)
...
 
 function march()
-    if Sys.ARCH == :aarch64 || Sys.ARCH == :arm
-        return (Int == Int32 ? `-march=armv7-a` : `-march=armv8-a+crypto+simd`)
+    if Sys.ARCH === :i686
+        return "-march=pentium4"
+    elseif Sys.ARCH === :x86_64
+        return "-march=x86-64"
+    elseif Sys.ARCH === :arm
+        return "-march=armv7-a+simd"
```
Comment (terasakisatoshi) [inline src/PackageCompilerX.jl:52]: - I could confirm `-march=armv7-a+simd` works fine for building `HelloX.jl` on my Raspberry Pi locally.

Comment (KristofferC) [reply src/PackageCompilerX.jl:52]: Thanks!

Comment (staticfloat) [reply src/PackageCompilerX.jl:52]: So if we use `+simd`, you can see in [the `-march=name` table on this page](https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html) we enable VFPv3 and NEON.  If you look at [the Raspberry PI FPU table on wikipedia](https://en.wikipedia.org/wiki/Raspberry_Pi#Specifications), and we only pay attention to the Model B, you'll see that is sufficient for the 2B and onward.  IMO, that's good enough, so let's just use this.  It's going to be really painful to use Julia without those options, and we don't have great BB support for manipulating these flags, so this should be a realistically conservative set of options.

File: src/PackageCompilerX.jl [hunk 2]
```diff
@@ -19,18 +34,28 @@ all_stdlibs() = readdir(Sys.STDLIB)
...
+    elseif Sys.ARCH === :aarch64
+        return "-march=armv8-a+crypto+simd"
+    elseif Sys.ARCH === :ppc64le # is this correct?
```
Comment (staticfloat) [inline src/PackageCompilerX.jl]: `:powerpc64le`, sadly.

File: test/runtests.jl [hunk 3]
```diff
@@ -11,6 +11,12 @@ cp(joinpath(DEPOT_PATH[1], "registries", "General"), joinpath(new_depot, "regist
 ENV["JULIA_DEPOT_PATH"] = new_depot
 Base.init_depot_path()
 
+is_slow_ci = haskey(ENV, "CI") && Sys.ARCH == :aarch64
```
Comment (staticfloat) [inline test/runtests.jl:14]: so sad

**Comment by KristofferC** [comment]
Something is messed up with the ARM builder but since https://github.com/JuliaComputing/PackageCompilerX.jl/pull/105#discussion_r376728342 confirmed this working, let's assume this is ok.

**Pull request merged**